core/src/test/java/com/linecorp/armeria/internal/common/PathAndQueryTest.java

```diff
@@ -544,4 +544,16 @@ private static PathAndQuery parse(@Nullable String rawPath, boolean allowDoubleD
         }
         return res;
     }
+
+    @Test
+    void assertSquareBracketsInPath() {
+                final PathAndQuery res = parse("/#/:@[]!$&'()*+,;=");
+        assertThat(res).isNotNull();
+        assertThat(res.path()).isNotEqualTo("/#/:@!$&'()*+,;=");
+
+        final PathAndQuery res2 =
+                parse("/%23%2F%3A%5B%5D%40%21%24%26%27%28%29%2A%2B%2C%3B%3D%3F");
```

Comment:
What is this for? Could you explain about this test please?

Reply:
The test is based off your comment here: https://github.com/line/armeria/pull/4615#pullrequestreview-1243317995

Reply:
Thanks for the [change](https://github.com/line/armeria/pull/4615/commits/1cca6e328ca7c3be4b1dcdf9de4706b9b76642ba#diff-5a50f1ba78c3d521cef86c09b4714219dd59c7ce1747b952f665ce411440e35bR557)! Now I get what it means. 🙇

Reply:
I get you too now :)